Lesson18/bilge/script.js

```diff
@@ -0,0 +1,209 @@
+  }
+
+  const avg = sum1 / array.length; // changed 'let' to 'const' as suggested //
+  return avg;
+}
+
+console.log("Exercise 2: ");
+
+console.log(findAvgOfAllElements([1, 2, 3, 4]));
+
+
+// 3. Write a function that takes an array of strings and concatenates them together adding spaces.
+// Input: ["Hello", "World", "!"]
+// Output: "Hello World !"
+
+console.log("Exercise 3:");
+
+let stringE3 = ["Hello", "World", "!"];
+
+console.log(stringE3.join(' ').replace(/ !/, '!'));   //I took the solution one step ahead by using "replace" method to eliminate the space before the exclamation mark as you suggested in the lesson// 
+
+
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+let arrayEx = [1, 2, 3, 4, 7, 10];
+
+function findNewArrayWithOddNumbers (array) {  
+  let newArr = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 !== 0) {
+      newArr.push(array[i]);  //Used push() method to add each odd number in the new array
+    }    
+  } 
+  return newArr;
+}
+
+console.log("Exercise 4: ");
+
+console.log(findNewArrayWithOddNumbers(arrayEx));
+
+
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10]
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+
+function findNewArrayWithEvenNumbers (array) {  
+  let newArr2 = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 === 0) {
+      newArr2.push(array[i]);  
+    }    
+  } 
+  return newArr2;
+}
+
+console.log('Exercise 5: ');
+
+console.log(findNewArrayWithEvenNumbers(testArrayEx5));
+
+
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+
+function getNewArray(array, removedValue) {
+  let resultArr = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] !== removedValue) {
+      resultArr.push(array[i]);
+    }
+  }
+  return resultArr;
+}
+
+console.log("Exercise 6: ");
+
+console.log(getNewArray(testArrayEx6, 2));
+
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+
+function getObjectKeys (object) {
+  let objectKeys = [];
+  for (let key in object) {
+    objectKeys.push(key);
+  }
+  return objectKeys;
+}
+
+console.log("Exercise 7: ");
+
+console.log(getObjectKeys(testObjEx7));
+
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = { name: "John", age: 30, city: "New York" };
+
+function findValuesOfObject (object) {
```

Comment:
There's also a built in method `Object.values(yourObject)` that returns all values of an object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values

No need to change current implementation.